src/Cookie.php

```diff
@@ -0,0 +1,494 @@
+        $new->maxAge = is_int($expiration) ? $expiration : null;
+        $new->expires = $this->normalizeExpires($expiration);
+
+        return $new;
+    }
+
+    /**
+     * Returns the expiration time.
+     *
+     * @return \DateTime|null
+     */
+    public function getExpires()
+    {
+        return $this->expires;
+    }
+
+    /**
+     * Checks if there is an expiration time.
+     *
+     * @return bool
+     */
+    public function hasExpires()
+    {
+        return isset($this->expires);
+    }
+
+    /**
+     * Sets the expires.
+     *
+     * @param \DateTime|null $expires
+     *
+     * @return self
+     */
+    public function withExpires(\DateTime $expires)
+    {
+        $new = clone $this;
+        $new->expires = $expires;
+
+        return $new;
+    }
+
+    /**
+     * Checks if the cookie is expired.
+     *
+     * @return bool
+     */
+    public function isExpired()
+    {
+        return isset($this->expires) and $this->expires < new \DateTime();
+    }
+
+    /**
+     * Returns the domain.
+     *
+     * @return string|null
+     */
+    public function getDomain()
+    {
+        return $this->domain;
+    }
+
+    /**
+     * Checks if there is a domain.
+     *
+     * @return bool
+     */
+    public function hasDomain()
+    {
+        return isset($this->domain);
+    }
+
+    /**
+     * Sets the domain.
+     *
+     * @param string|null $domain
+     *
+     * @return self
+     */
+    public function withDomain($domain)
+    {
+        $new = clone $this;
+        $new->domain = $this->normalizeDomain($domain);
+
+        return $new;
+    }
+
+    /**
+     * Matches a domain.
+     *
+     * @param string $domain
+     *
+     * @return bool
+     *
+     * @see http://tools.ietf.org/html/rfc6265#section-5.1.3
+     */
+    public function matchDomain($domain)
+    {
+        // Domain is not set or exact match
+        if (!$this->hasDomain() || strcasecmp($domain, $this->domain) === 0) {
+            return true;
+        }
+
+        // Domain is not an IP address
+        if (filter_var($domain, FILTER_VALIDATE_IP)) {
+            return false;
+        }
+
+        return (bool) preg_match('/\b'.preg_quote($this->domain).'$/i', $domain);
+    }
+
+    /**
+     * Returns the path.
+     *
+     * @return string
+     */
+    public function getPath()
+    {
+        return $this->path;
+    }
+
+    /**
+     * Sets the path.
+     *
+     * @param string|null $path
+     *
+     * @return self
+     */
+    public function withPath($path)
+    {
+        $new = clone $this;
+        $new->path = $this->normalizePath($path);
+
+        return $new;
+    }
+
+    /**
+     * It matches a path.
+     *
+     * @param string $path
+     *
+     * @return bool
+     *
+     * @see http://tools.ietf.org/html/rfc6265#section-5.1.4
+     */
+    public function matchPath($path)
+    {
+        return $this->path === $path || (strpos($path, $this->path.'/') === 0);
```

Comment:
Checked it. Yes, the trailing slash is necessary.

http://tools.ietf.org/html/rfc6265#section-5.1.4
http://tools.ietf.org/html/rfc6265#section-5.2.4